modules/capi-yaml-machine-template-vsphere.adoc

```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cluster_api_machine_management/cluster_api_provider_configurations/cluster-api-config-options-vsphere.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="capi-yaml-machine-template-vsphere_{context}"]
+= Sample YAML for a Cluster API machine template resource on {vmw-full}
+
+The machine template resource is provider-specific and defines the basic properties of the machines that a compute machine set creates.
+The compute machine set references this template when creating machines.
+
+[source,yaml]
+----
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VSphereMachineTemplate # <1>
+metadata:
+  name: <template_name> # <2>
+  namespace: openshift-cluster-api
+spec:
+  template:
+    spec: # <3>
+      template: <vm_template_name> # <4>
+      server: <vcenter_server_ip> # <5>
+      diskGiB: 128
+      cloneMode: linkedClone # <6>
+      datacenter: <vcenter_datacenter_name> # <7>
+      datastore: <vcenter_datastore_name> # <8>
+      folder: <vcenter_vm_folder_path> # <9>
+      resourcePool: <vsphere_resource_pool> # <10>
+      numCPUs: 4
+      memoryMiB: 16384
+      network:
+        devices:
+        - dhcp4: true
+          networkName: "<vm_network_name>" # <11>
+----
+<1> Specify the machine template kind.
+This value must match the value for your platform.
+<2> Specify a name for the machine template.
+<3> Specify the details for your environment.
+The values here are examples.
+<4> Specify the vSphere VM template to use, such as `user-5ddjd-rhcos`.
+<5> Specify the vCenter server IP or fully qualified domain name.
+<6> Specify the type of VM clone to use.
+The following values are valid:
++
+--
+* `fullClone`
+* `linkedClone`
+--
++
+When using the `linkedClone` type, the disk size matches the clone source instead of using the `diskGiB` value.
+For more information, see the {vmw-short} documentation about VM clone types.
```

Comment:
Would it be helpful to have an additional resource section for the assembly and provide a link, or is it not worth it?